File: FarmLand-Backend/helpers/schedule.py
from dataclasses import dataclass
from typing import Optional
from datetime import datetime

@dataclass
class ScheduleHelper:
    farm_id: int
    days_after_sowing: int
    fertilizer_type: str
    quantity: float
    quantity_unit: str
    id: Optional[int] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Errors:
        SCHEDULE_NOT_FOUND = {
            "msg": "Schedule not found",
            "status_code": 404,
            "additional_info": {
                "error_type": "CLIENT",
                "error_name": "SCHEDULE_NOT_FOUND"
            }
        }

    def validate(self) -> bool:
        if not self.fertilizer_type or self.quantity <= 0 or self.days_after_sowing < 0:
            raise Exception("Invalid schedule data")
        return True

File: farmland-frontend/src/app/interfaces/schedule.ts
export interface Schedule {
  id: number;
  farm_id: number;
  days_after_sowing: number;
  fertilizer_type: string;
  quantity: number;
  quantity_unit: string;
}

export interface DueSchedules {
  today_tomorrow: Schedule[];
}

export interface ScheduleResponse {
  data: {
    today_tomorrow: Schedule[];
  };
}
